app/javascript/src/Users/permissions.jsx

```diff
@@ -184,10 +184,10 @@ export const ServiceAccess = { // eslint-disable-line no-unused-vars
   render ({props, context, dispatch}) {
     let { id, name } = props.service || {}
     let all = !(new Set(context.admin_sections)).has('services')
-    let ids = context.member_permission_service_ids
+    let ids = context.member_permission_service_ids || []
 
-    let checked = all || !ids || ids.includes(id)
-    let disabled = !ids || all
+    let checked = all || ids.includes(id)
+    let disabled = all
```

Comment:
What does `disabled` and `all` exactly means? And why are they equal?

Reply:
`disabled` means that the checkboxes are disabled. `all` - that all services are accessible.
They are equal because when all services are accessible, you cannot select specific services, hence the checkboxes are disabled.
<img width="896" alt="all_services" src="https://user-images.githubusercontent.com/1270328/49364330-555dff00-f6e4-11e8-8816-cfd4dd318691.png">
 
Maybe it can be refactored and `disabled` variable removed, but I think it helps understand the behavior :)

Reply:
Ok I think I get what you're saying. It looks good as it reads `disabled = all # all checkboxes disabled` but code-wise it does not make that much sense. Naming is confusing, _all_ kinda implies it is some sort of collection, whereas _disabled_ seems like a boolean.
`!(new Set(context.admin_sections)).has('services')` is a bit convoluted but I think it means something like admin don't have services..? And that's called _all_. Also, the component we have here has only 1 checkbox, so why this checkbox is disabled and checked when _all_?

Reply:
@josemigallas Regarding `!(new Set(context.admin_sections)).has('services')`: well, this is because of the feature design :) See the description: when only **some** services are enabled, user has a `MemberPermission` with `admin_section: services` and `service_ids: [<comma-separated-list>]`. When all services are disabled, the user will also have the `MemberPermission` with `admin_section: services`, this time with `service_ids: []`.

But when the user doesn't have any services limitations (_"All current and future APIs"_ enabled), the user **does not** have the `MemberPermission` with `admin_section: services` :)

So, in this case `all == true`, meaning "all services are enabled", and hence `disabled == true`, meaning that the checkboxes for components for each individual service (this is what `ServiceAccess` is responsible for) is disabled.

Is it more clear now? :)
Would it help to have it like this?

~~~
...
    # if the user doesn't have a MemberPermission with `services` admin section, it means all APIs are enabled
    let all_services_enabled = !(new Set(context.admin_sections)).has('services')
    let ids = context.member_permission_service_ids || []

    let checked = all_services_enabled || ids.includes(id)
    # if all services are enabled, individual checkboxes for each service should be disabled
    let disabled = all_services_enabled
...
~~~

Or we could get rid of the `disabled` variable, but then this would be confusing:
~~~
<input ... disabled={ all_services_enabled } .../>
~~~

or not? :)